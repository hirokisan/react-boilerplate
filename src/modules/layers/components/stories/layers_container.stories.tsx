import React from 'react';

import { LayersContainer } from '../layers_container';

export default {
  title: 'Layers'
};

const Wrapper = () => <LayersContainer>werwer 12012</LayersContainer>;

export const LayerContainer = () => <Wrapper />;

LayerContainer.story = {
  name: 'default'
};
