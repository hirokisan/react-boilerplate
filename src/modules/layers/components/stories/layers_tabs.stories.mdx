import { action } from '@storybook/addon-actions';
import { Story, Meta, Props, Source } from '@storybook/addon-docs/blocks';
import { text } from '@storybook/addon-knobs';
import { LayersTabs } from '../layers_tabs';
import { array, number } from '@storybook/addon-knobs';
import { LayersWrapper } from '../layers_container';

<Meta 
  title='Components|Layers/Tabs' 
  component={LayersTabs} 
  decorators={[(storyFn) => <LayersWrapper>{storyFn()}</LayersWrapper>]} 
/>

# Layer Tabs

Layer tabs switch between different layer groups

### Example

<Story name="first selected">
  <LayersTabs
    currentTab={number('ActiveTab', 0)}
    setTab={action('setTab')}
    tabs={array('Tabs', ['LAYERS', 'INDICATORS'])}
  />
</Story>

### Props

<Props of={LayersTabs} />

## Stories

### Second Selected

<Story name="second selected">
  <LayersTabs
    currentTab={1}
    setTab={action('setTab')}
    tabs={['LAYERS', 'INDICATORS']}
  />
</Story>