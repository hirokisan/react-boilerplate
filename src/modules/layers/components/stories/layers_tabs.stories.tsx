import React from 'react';
import { action } from '@storybook/addon-actions';
import { LayersTabs } from '../layers_tabs';
import { array, number } from '@storybook/addon-knobs';
import { LayersWrapper } from '../layers_container';
import { Action } from 'config/common';

export default {
  title: 'Components|Layers/Tabs',
  parameters: {
    component: LayersTabs,
    // eslint-disable-next-line react/display-name
    decorators: [(storyFn: Function) => <LayersWrapper>{storyFn()}</LayersWrapper>]
  }
};

const storyProps = {
  setTab: action('setTab') as Action<number>
};

export const DefaultView = ({ setTab } = storyProps) => (
  <LayersTabs
    currentTab={number('ActiveTab', 0)}
    setTab={setTab}
    tabs={array('Tabs', ['LAYERS', 'INDICATORS'])}
  />
);

DefaultView.story = {
  name: 'first selected'
};

export const SelectedView = ({ setTab } = storyProps) => (
  <LayersTabs currentTab={1} setTab={setTab} tabs={['LAYERS', 'INDICATORS']} />
);

SelectedView.story = {
  name: 'second selected'
};
