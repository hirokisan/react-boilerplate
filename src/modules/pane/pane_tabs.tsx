import React, { FC } from 'react';

import { classes, styled, Action, numberAttribute, theme } from 'config/common';

const TabsContainer = styled.div`
  width: 100%;
  height: 40px;
  min-height: 40px;
  display: flex;
  align-items: center;
  justify-content: center;
`;

const TabsList = styled.ul`
  width: 100%;
  height: 100%;
  align-items: center;
  display: flex;
  justify-content: flex-start;
`;

const TabItem = styled.li`
  height: 100%;
  align-items: center;
  color: rgba(255, 255, 255, 0.4);
  border-left: 1px solid ${theme.paneBackground};
  border-right: 1px solid ${theme.divider};
  cursor: pointer;
  display: flex;
  font-size: 11px;
  font-weight: 400;
  justify-content: center;
  line-height: 15px;
  padding: 0 12px;
  transition: background-color 0.2s ease-out;

  &:hover,
  &.active {
    background-color: rgba(77, 77, 77, 0.17);
    border-left-color: ${theme.divider};
  }
  &.active {
    border-right-color: transparent;
    border-left: 1px solid rgba(77, 77, 77, 0.17);
    color: ${theme.paneText};
  }
`;

const Tab = styled.div`
  display: none;
  &.active {
    display: block;
  }
`;

type Props = {
  /**
   * List of tabs to display
   */
  tabs: string[];
};

const defaultTabs = ['LAYERS', 'INDICATORS'];

export const PaneTabs: FC<Props> = ({ children, tabs = defaultTabs }) => {
  const [currentTab, setTab] = React.useState(0);

  const setActiveTab = React.useCallback(
    (e: React.MouseEvent<HTMLLIElement>) => {
      setTab(numberAttribute(e.currentTarget, 'data-tab'));
    },
    [setTab]
  );

  return (
    <div>
      <TabsContainer>
        <TabsList>
          {tabs.map((tab, i) => (
            <TabItem
              key={i}
              className={classes({ active: currentTab === i })}
              data-tab={i}
              onClick={setActiveTab}
            >
              {tab}
            </TabItem>
          ))}
        </TabsList>
      </TabsContainer>
      {children &&
        Array.isArray(children) &&
        children.map((child, i) => (
          <Tab key={i} className={classes({ active: currentTab === i })}>
            {child}
          </Tab>
        ))}
    </div>
  );
};
