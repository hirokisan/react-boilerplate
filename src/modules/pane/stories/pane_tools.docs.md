# Title

This is some text in an advanced Markdown specification
