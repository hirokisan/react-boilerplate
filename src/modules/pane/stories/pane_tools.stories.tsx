import React from 'react';

import { action } from '@storybook/addon-actions';
import { ToolsContainer } from '../pane_tools';
import { PaneContainer } from '../pane_container';

import docs from './pane_tools.docs.md';
import marked from 'marked';
import { Action } from 'config/common';

export default {
  title: 'Components|Pane/Tools',
  parameters: {
    component: ToolsContainer,
    info: marked(docs),
    // eslint-disable-next-line react/display-name
    decorators: [(storyFn: Function) => <PaneContainer>{storyFn()}</PaneContainer>]
  }
};

export const DefaultView = (setToolHandler?: Action<string>) => (
  <ToolsContainer setTool={setToolHandler || action('setTool')} />
);

DefaultView.story = {
  name: 'default'
};
