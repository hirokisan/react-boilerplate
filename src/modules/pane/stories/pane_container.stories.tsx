import React from 'react';

import { ToolsContainer } from '../pane_tools';
import { PaneContainer } from '../pane_container';

import marked from 'marked';
import { PaneTabs } from '../pane_tabs';
import { action } from '@storybook/addon-actions';

const docs = `
The layer container is one of the core system components, allowing user to display and manipulate layers. 
`;

export default {
  title: 'Components|Pane/Container',
  parameters: {
    component: ToolsContainer,
    info: marked(docs)
  }
};

export const DefaultView = () => <PaneContainer>This is some content</PaneContainer>;

DefaultView.story = {
  name: 'default'
};

const Story = () => {
  const [tab, setTab] = React.useState(0);
  return (
    <PaneContainer>
      <PaneTabs setTab={setTab} currentTab={tab} tabs={['LAYERS', 'ACTIONS']} />
      <ToolsContainer setTool={action('setTool')} />
      <div style={{ padding: '6px', color: 'white' }}>
        {tab == 0 && <div>Tab LAYERS</div>}
        {tab == 1 && <div>Tab ACTIONS</div>}
      </div>
    </PaneContainer>
  );
};

// weird bug not allowing to render hooks directly in the story
// we just use a sub component
export const CompoundView = () => {
  return <Story />;
};

CompoundView.story = {
  name: 'compound view'
};
