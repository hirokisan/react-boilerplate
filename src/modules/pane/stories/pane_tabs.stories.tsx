import React from 'react';
import { action } from '@storybook/addon-actions';
import { PaneTabs } from '../pane_tabs';
import { array, number } from '@storybook/addon-knobs';
import { PaneContainer } from '../pane_container';
import { Action } from 'config/common';

export default {
  title: 'Components|Pane/Tabs',
  parameters: {
    component: PaneTabs,
    // eslint-disable-next-line react/display-name
    decorators: [(storyFn: Function) => <PaneContainer>{storyFn()}</PaneContainer>]
  }
};

const storyProps = {
  setTab: action('setTab') as Action<number>
};

export const DefaultView = ({ setTab } = storyProps) => (
  <PaneTabs
    currentTab={number('ActiveTab', 0)}
    setTab={setTab}
    tabs={array('Tabs', ['LAYERS', 'INDICATORS'])}
  />
);

DefaultView.story = {
  name: 'first selected'
};

export const SelectedView = ({ setTab } = storyProps) => (
  <PaneTabs currentTab={1} setTab={setTab} tabs={['LAYERS', 'INDICATORS']} />
);

SelectedView.story = {
  name: 'second selected'
};
