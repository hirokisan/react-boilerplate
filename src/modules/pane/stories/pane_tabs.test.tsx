import React from 'react';
import { render, fireEvent, getByText } from '@testing-library/react';

import { DefaultView, SelectedView } from './pane_tabs.stories';

describe('Layer Tabs', () => {
  it('renders default', () => {
    const { container } = render(DefaultView({ setTab: jest.fn() }));
    expect(container).toMatchSnapshot();
  });

  it('changes active tab by index', () => {
    const { getByText } = render(SelectedView({ setTab: jest.fn() }));
    expect(getByText('INDICATORS')).toHaveClass('active');
  });

  it('calls the change tab function', () => {
    const setTab = jest.fn();
    const { getByText } = render(SelectedView({ setTab }));

    fireEvent.click(getByText('INDICATORS'));
    expect(setTab).toHaveBeenCalledWith(1);
  });
});
