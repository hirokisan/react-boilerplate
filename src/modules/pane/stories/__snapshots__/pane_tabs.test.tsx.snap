// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layer Tabs renders default 1`] = `
.emotion-3 {
  width: 100%;
  height: 40px;
  min-height: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-2 {
  width: 100%;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.emotion-0 {
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: rgba(255,255,255,0.4);
  border-left: 1px solid #1F212B;
  border-right: 1px solid #000000;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 11px;
  font-weight: 400;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 15px;
  padding: 0 12px;
  -webkit-transition: background-color 0.2s ease-out;
  transition: background-color 0.2s ease-out;
}

.emotion-0:hover,
.emotion-0.active {
  background-color: rgba(77,77,77,0.17);
  border-left-color: #000000;
}

.emotion-0.active {
  border-right-color: transparent;
  border-left: 1px solid rgba(77,77,77,0.17);
  color: #FFFFFF;
}

<div>
  <div
    class="emotion-3"
  >
    <ul
      class="emotion-2"
    >
      <li
        class="active emotion-0"
        data-tab="0"
      >
        LAYERS
      </li>
      <li
        class=" emotion-0"
        data-tab="1"
      >
        INDICATORS
      </li>
    </ul>
  </div>
</div>
`;
